test/broadcast.jl

```diff
@@ -165,9 +165,9 @@ m = [1:2;]'
 @test @inferred([0,1.2].+reshape([0,-2],1,1,2)) == reshape([0 -2; 1.2 -0.8],2,1,2)
 rt = Base.return_types(.+, Tuple{Array{Float64, 3}, Array{Int, 1}})
 @test length(rt) == 1 && rt[1] == Array{Float64, 3}
-rt = Base.return_types(broadcast, Tuple{Function, Array{Float64, 3}, Array{Int, 1}})
+rt = Base.return_types(broadcast, Tuple{typeof(+), Array{Float64, 3}, Array{Int, 1}})
```

Comment:
This is another place where the return type is assumed to be equal to `promote_type` on the inputs. It breaks with the `promote_op` definition based on `typeof(op(one(R), one(S)))`, since the exact function needs to be specified to call it.